R/fit_baseline_models.R

```diff
@@ -0,0 +1,69 @@
+#' Generate predictions for all baseline models for the given reference date
+#'
+#' @inheritParams fit_baselines_one_location
+#' @param target character string specifying the name of the prediction target
+#'
+#' @details The `model_variations` data frame has the following columns and
+#'   possible values for each:
+#'   - transformation (character): "none" or "sqrt", determines distribution shape
+#'   - symmetrize (boolean), determines if distribution is symmetric
+#'   - window_size (integer), determines how many previous observations inform
+#'     the forecast
+#'
+#' @return `model_out_tbl` of forecasts for all baseline models with columns:
+#'   `model_id`, `reference_date`, `location`, `horizon`, `target`,
+#'   `target_end_date`, `output_type`, `output_type_id`, and `value`
+#'
+#' @importFrom rlang .data
+#'
+fit_baseline_models <- function(model_variations,
+                                target_ts,
+                                reference_date,
+                                temporal_resolution,
+                                horizons,
+                                target,
+                                quantile_levels,
+                                n_samples,
+                                round_predictions = FALSE,
+                                seed = NULL) {
+  if (is.null(target)) {
+    cli::cli_abort("{.arg target} is missing; please provide one")
+  }
+
+  # fit baseline models
+  #furrr::future_pmap_dfr(
```

Comment:
I commented out this line because I wanted some feedback on whether to use `map()` or the `furrr` equivalent like in the original code. Fewer dependencies may be better but calculating the forecasts can take a while. Thoughts?

Reply:
I'm honestly not sure — I haven't done any tests comparing the two. Maybe the best thing to do is stick with `purrr::map` for now and if later testing reveals it's too slow I can switch back over to `furrr`

Reply:
sounds good